diff --git a/kubernetes/src/main/java/io/kubernetes/client/openapi/models/V1ListMeta.java b/kubernetes/src/main/java/io/kubernetes/client/openapi/models/V1ListMeta.java
index f161284a2..d3d563bbb 100644
--- a/kubernetes/src/main/java/io/kubernetes/client/openapi/models/V1ListMeta.java
+++ b/kubernetes/src/main/java/io/kubernetes/client/openapi/models/V1ListMeta.java
@@ -258,7 +258,8 @@ public class V1ListMeta {
            @Override
            public V1ListMeta read(JsonReader in) throws IOException {
              JsonElement jsonElement = elementAdapter.read(in);
-             validateJsonElement(jsonElement);
+             // Disable validation so delete API can tolerate non-status return object (graceful deletion)
+             // validateJsonObject(jsonObj);
              return thisAdapter.fromJsonTree(jsonElement);
            }
 
