diff --git a/kubernetes/src/main/java/io/kubernetes/client/openapi/models/V1Secret.java b/kubernetes/src/main/java/io/kubernetes/client/openapi/models/V1Secret.java
index 5684f4776..aa77bd40f 100644
--- a/kubernetes/src/main/java/io/kubernetes/client/openapi/models/V1Secret.java
+++ b/kubernetes/src/main/java/io/kubernetes/client/openapi/models/V1Secret.java
@@ -18,6 +18,7 @@ import com.google.gson.annotations.JsonAdapter;
 import com.google.gson.annotations.SerializedName;
 import com.google.gson.stream.JsonReader;
 import com.google.gson.stream.JsonWriter;
+import io.kubernetes.client.custom.MapUtils;
 import io.kubernetes.client.openapi.models.V1ObjectMeta;
 import java.io.IOException;
 import java.util.Arrays;
@@ -243,8 +244,8 @@ public class V1Secret implements io.kubernetes.client.common.KubernetesObject {
       return false;
     }
     V1Secret v1Secret = (V1Secret) o;
-    return Objects.equals(this.apiVersion, v1Secret.apiVersion) &&
-        Objects.equals(this.data, v1Secret.data) &&
+    return Objects.equals(this.apiVersion, v1Secret.apiVersion) &&
+        MapUtils.equals(this.data, v1Secret.data) &&
         Objects.equals(this.immutable, v1Secret.immutable) &&
         Objects.equals(this.kind, v1Secret.kind) &&
         Objects.equals(this.metadata, v1Secret.metadata) &&
