diff --git a/kubernetes/src/main/java/io/kubernetes/client/openapi/models/V1Secret.java b/kubernetes/src/main/java/io/kubernetes/client/openapi/models/V1Secret.java
index 8fdadaac6..5fe296242 100644
--- a/kubernetes/src/main/java/io/kubernetes/client/openapi/models/V1Secret.java
+++ b/kubernetes/src/main/java/io/kubernetes/client/openapi/models/V1Secret.java
@@ -19,6 +19,7 @@ import com.google.gson.annotations.JsonAdapter;
 import com.google.gson.annotations.SerializedName;
 import com.google.gson.stream.JsonReader;
 import com.google.gson.stream.JsonWriter;
+import io.kubernetes.client.custom.MapUtils;
 import io.kubernetes.client.openapi.models.V1ObjectMeta;
 import java.io.IOException;
 import java.util.HashMap;
@@ -259,7 +260,7 @@ public class V1Secret implements io.kubernetes.client.common.KubernetesObject {
     }
     V1Secret v1Secret = (V1Secret) o;
     return Objects.equals(this.apiVersion, v1Secret.apiVersion) &&
-        Objects.equals(this.data, v1Secret.data) &&
+        MapUtils.equals(this.data, v1Secret.data) &&
         Objects.equals(this.immutable, v1Secret.immutable) &&
         Objects.equals(this.kind, v1Secret.kind) &&
         Objects.equals(this.metadata, v1Secret.metadata) &&
