diff --git a/kubernetes/src/main/java/io/kubernetes/client/openapi/models/V1Status.java b/kubernetes/src/main/java/io/kubernetes/client/openapi/models/V1Status.java
index b529e3455..c2268419a 100644
--- a/kubernetes/src/main/java/io/kubernetes/client/openapi/models/V1Status.java
+++ b/kubernetes/src/main/java/io/kubernetes/client/openapi/models/V1Status.java
@@ -378,7 +378,8 @@ public class V1Status {
            @Override
            public V1Status read(JsonReader in) throws IOException {
              JsonElement jsonElement = elementAdapter.read(in);
-             validateJsonElement(jsonElement);
+             // Disable validation so delete API can tolerate non-status return object (graceful deletion)
+             // validateJsonObject(jsonObj);
              return thisAdapter.fromJsonTree(jsonElement);
            }
 
